components/can/Can.tsx

```diff
@@ -0,0 +1,41 @@
+import styles from "../can/Can.module.css";
+
+export type CanProps = {
+  type: "paper" | "recycle" | "residual";
+  className: string;
```

Comment:
This might be confusing because of the html property `classNames`. 
In your case, you use `className` to read  a specific module className. I understand, that this sounds very similar.

Still, I would change this to make sure that not every `className` is possible. E.g. you couldn't say `className="box"`.
Instead, I would limit the options with a union like `className: 'canresidual' | 'canrecycle ' | 'canpaper '`.
If you do this, you will see that this is very similar to `type`.

Actually, you don't need `className` property at all because it will always depend on `type` value.
If `type` is `paper`, you would say that `className` is `canpaper`. 

That's why I would recommend removing `className` and change the class names in `Can.module.css` to this:

```css
.residual {
  background: url(/canresidual.png);
}
.recycle {
  background: url(/canrecycling.png);
}
.paper {
  background: url(/canpaper.png);
}
```

Now, you can change your diff from `className={styles[className]}` to `className={styles[type]}`